sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java

```diff
@@ -0,0 +1,262 @@
+        return new DefaultSyncPoller<>(this.defaultPollInterval,
+            this.activationOperation,
+            this.pollOperation,
+            this.cancelOperation,
+            this.fetchResultOperation);
+    }
+
+    /**
+     * Returns a decorated Mono, upon subscription it internally subscribes to the Mono that perform one
+     * time activation. The decorated Mono caches the result of activation operation in a PollResponse,
+     * this cached response will be replayed for any future subscriptions.
+     *
+     * Note: we can't use the reactor standard cache() operator, because it caches error terminal signal
+     * and forward it to any future subscriptions. If there is an error from activation Mono
+     * then we don't want to cache it but just forward it to subscription that initiated the failed activation.
+     * For any future subscriptions we don't want to forward the past error instead activation
+     * should again invoked. Once a subscription received a successful event from activation Mono then we want
+     * to cache it and replay it to any future subscriptions.
+     *
+     * The decorated Mono also handles concurrent calls to activation. Only one of them will be able to call
+     * activation and other subscriptions will keep resubscribing until it sees a cached response or get a chance
+     * to call activation as the one previously entered the critical section got an error on activation.
+     *
+     * @param activationOperation a supplier that provide a Mono upon subscription execute activation operation.
+     * @return
+     */
+    private Mono<PollResponse<T>> deferredActivationMono(final Supplier<Mono<T>> activationOperation) {
+        return Mono.defer(() -> {
+            if (this.activationResponse != null) {
+                return Mono.just(this.activationResponse);
+            }
+            if (this.guardActivationCall.compareAndSet(this, 0, 1)) {
+                final Mono<T> activationMono;
+                try {
+                    activationMono = activationOperation.get();
+                } catch (Throwable throwable) {
+                    return Mono.error(throwable);
+                }
+                return activationMono.map((T result) -> {
+                    this.activationResponse = new PollResponse<>(LongRunningOperationStatus.NOT_STARTED, result);
+                    return this.activationResponse;
+                })
+                .switchIfEmpty(Mono.defer(() -> {
+                    this.activationResponse = new PollResponse<>(LongRunningOperationStatus.NOT_STARTED, null);
+                    return Mono.just(this.activationResponse);
+                }))
+                .doOnError(throwable -> guardActivationCall.compareAndSet(this, 1, 0));
+            } else {
+                return Mono.empty();
+            }
+        }).repeatWhenEmpty((Flux<Long> longFlux) -> longFlux.concatMap(ignored -> Flux.just(true)));
```

Comment:
Should there be a limit to how many times/how long this repeat should run? If the `activationOperation.get()` takes a long time, this might spin for a bit which can cause high CPU usage.

Reply:
busy loop runs only on the rare occurring concurrency cases, like user using the same PollerFlux instance in merge/flatmap operator concurrently. It should be fine to busy loop here, its very common pattern to handle such edge cases.